exchange/bitswap/decision/engine.go

```diff
@@ -298,6 +298,22 @@ func (e *Engine) findOrCreate(p peer.ID) *ledger {
 	return l
 }
 
+type LedgerSnapshot struct {
+	DebtRatio     float64
+	BytesSent     uint64
+	BytesReceived uint64
+}
+
+func (e *Engine) LedgerSnapshot(p peer.ID) *LedgerSnapshot {
+	snapshot := &LedgerSnapshot{
+		DebtRatio:     e.findOrCreate(p).Accounting.Value(),
```

Comment:
I think we're going to have to worry about locking a bit here. Lets make a single call to `findOrCreate` which is threadsafe on its own, then call lock on the returned ledger, defer the unlock, and then take the information we need from it